kestrel-opentelemetry-metrics/src/main/kotlin/com/dreweaster/ddd/kestrel/application/reporting/opentelemetry/metrics/OpenTelemetryMetricsDomainModelReporter.kt

```diff
@@ -0,0 +1,237 @@
+package com.dreweaster.ddd.kestrel.application.reporting.opentelemetry.metrics
+
+import com.dreweaster.ddd.kestrel.application.*
+import com.dreweaster.ddd.kestrel.domain.Aggregate
+import com.dreweaster.ddd.kestrel.domain.AggregateState
+import com.dreweaster.ddd.kestrel.domain.DomainCommand
+import com.dreweaster.ddd.kestrel.domain.DomainEvent
+import io.opentelemetry.api.OpenTelemetry
+import io.opentelemetry.api.common.AttributeKey.stringKey
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.api.metrics.LongCounter
+import io.opentelemetry.api.trace.SpanKind
+import io.opentelemetry.api.trace.Tracer
+
+class OpenTelemetryMetricsDomainModelReporter(openTelemetry: OpenTelemetry) : DomainModelReporter {
+    val aggregateTypeKey = "aggregate_type"
+    val eventTypeKey = "event_type"
+    val commandTypeKey = "command_type"
+    val resultKey = "result"
+    val deduplicatedKey = "deduplicated"
+
+    private val tracer: Tracer = openTelemetry.getTracer("com.dreweaster.ddd.kestrel.DomainModelReporter")
+
+
+    val commandExecutionLatency = tracer.spanBuilder("aggregate_command_execution_latency_seconds")
+        .setSpanKind(SpanKind.INTERNAL)
+
+    var meter = openTelemetry.meterBuilder("com.dreweaster.ddd.kestrel.DomainModelReporter")
+        .build()
+
+    var commandExecution: LongCounter = meter
+        .counterBuilder("aggregate_command_execution_total")
+        .setDescription("Total aggregate commands executed")
+        .setUnit("1")
+        .build()
+
+    var eventsEmitted: LongCounter = meter
+        .counterBuilder("aggregate_events_emitted_total")
+        .setDescription("Total aggregate events emitted")
+        .setUnit("1")
+        .build()
+
+    var aggregateRecovery: LongCounter = meter
+        .counterBuilder("aggregate_recovery_total")
+        .setDescription("Total aggregates recovered")
+        .setUnit("1")
+        .build()
+
+    var applyCommand: LongCounter = meter
+        .counterBuilder("aggregate_apply_command_total")
+        .setDescription("Total aggregate commands applied")
+        .setUnit("1")
+        .build()
+
+    var persistEvents: LongCounter = meter
+        .counterBuilder("aggregate_persist_events_total")
+        .setDescription("Total calls to persist events for aggregate")
+        .setUnit("1")
+        .build()
```

Comment:
_Minor_
Lots of things that looks like they should be immutable

```suggestion
    private val meter = openTelemetry.meterBuilder("com.dreweaster.ddd.kestrel.DomainModelReporter")
        .build()

    val commandExecution: LongCounter = meter
        .counterBuilder("aggregate_command_execution_total")
        .setDescription("Total aggregate commands executed")
        .setUnit("1")
        .build()

    val eventsEmitted: LongCounter = meter
        .counterBuilder("aggregate_events_emitted_total")
        .setDescription("Total aggregate events emitted")
        .setUnit("1")
        .build()

    val aggregateRecovery: LongCounter = meter
        .counterBuilder("aggregate_recovery_total")
        .setDescription("Total aggregates recovered")
        .setUnit("1")
        .build()

    val applyCommand: LongCounter = meter
        .counterBuilder("aggregate_apply_command_total")
        .setDescription("Total aggregate commands applied")
        .setUnit("1")
        .build()

    val persistEvents: LongCounter = meter
        .counterBuilder("aggregate_persist_events_total")
        .setDescription("Total calls to persist events for aggregate")
        .setUnit("1")
        .build()
```

Reply:
Done